README.md

```diff
@@ -1,76 +1,134 @@
-Ruby api client for [Auth0](https://auth0.com) platform
-
 [![Build Status](https://travis-ci.org/auth0/ruby-auth0.svg?branch=master)](https://travis-ci.org/auth0/ruby-auth0)
 [![Gem Version](https://badge.fury.io/rb/auth0.svg)](http://badge.fury.io/rb/auth0)
 [![Coverage Status](https://coveralls.io/repos/auth0/ruby-auth0/badge.svg?branch=master)](https://coveralls.io/r/auth0/ruby-auth0?branch=master)
-[![Dependency Status](https://gemnasium.com/auth0/ruby-auth0.svg)](https://gemnasium.com/auth0/ruby-auth0)
 [![Yard Docs](http://img.shields.io/badge/yard-docs-blue.svg)](http://www.rubydoc.info/github/auth0/ruby-auth0/master/frames)
 
+Ruby API client for the [Auth0](https://auth0.com) platform.
+
 ## Installation
 
+This gem can be installed directly:
+
 ``` bash
 $ gem install auth0
 ```
 
-or with [Bundler](http://bundler.io)
+... or with [Bundler](https://bundler.io/man/bundle-add.1.html):
 
-```ruby
-gem 'auth0'
+```bash
+bundle add auth0
 ```
 
-## Basic usage
+## API Documentation
 
-Using [APIv2](https://auth0.com/docs/api/v2)
+You can build the API documentation with the following:
 
-```ruby
-require "auth0"
+``` bash
+bundle exec rake documentation
+```
 
-auth0 = Auth0Client.new(
-  :client_id => "YOUR CLIENT ID",
-  :token => "YOUR JWT HERE",
-  :domain => "<YOUR ACCOUNT>.auth0.com",
-  :api_version => 2
-)
+To view the generated documentation, open `doc/Auth0/Api.html`
 
-puts auth0.get_users
-```
+## Management API v2
 
-### Timeout
-You can setup a custom timeout in the Auth0Client. By default it is set to 10 seconds.
+This SDK provides access to the [Management API v2](https://auth0.com/docs/api/management/v2) via modules that help create clear and accurate calls. Most of the interaction is done through the `Auth0Client` class, instantiated with the required credentials.
+
+As a simple example of how to get started, we'll create an admin route to point to a list of all users from Auth0:
 
 ```ruby
-require "auth0"
+# config/routes.rb
+Rails.application.routes.draw do
+  # ...
+  get 'admin/users', to: 'all_users#index'
+  # ...
+end
+```
 
-auth0 = Auth0Client.new(
-  :client_id => "YOUR CLIENT ID",
-  :token => "YOUR JWT HERE",
-  :domain => "<YOUR ACCOUNT>.auth0.com",
-  :timeout => 15
-)
+... and a Controller to handle that route:
 
-puts auth0.get_users
+```ruby
+# app/controllers/all_users_controllers.rb
+require 'auth0'
+
+class AllUsersController < ApplicationController
+	before_action :set_api
```

Comment:
Indentation should be 2 spaces in. This looks like it may be a tab. This may have happened when I copied the example to you from a shoddy copy out of my terminal :)

Reply:
Was there a reason you didn't go with the memoized API we chatted about being idiomatic Rails?

```ruby
class AllUsersController < ApplicationController
  # Get all users from Auth0 with "auth0" in their email.
  def index
   @params = {
     q: "email:*auth0*",
     fields: 'email,user_id,name',
     include_fields: true,
     page: 0,
     per_page: 50
   }
   @users = auth0.users @params
  end

  private

  # Setup the Auth0 API connection.
  def auth0
   @auth0 ||= Auth0Client.new(
     client_id: ENV['AUTH0_RUBY_CLIENT_ID'],
     token: ENV['AUTH0_RUBY_API_TOKEN'],
     domain: ENV['AUTH0_RUBY_DOMAIN'],
     api_version: 2,
     timeout: 15 # optional, defaults to 10s
   )
  end
end
```

This will allow a user to reference `auth0` method as many times as they want without having to establish a new connection or concern themselves with setting it up in a `before_action`.